srcts/src/components/card.ts

```diff
@@ -93,6 +113,7 @@ class Card {
     // Let Shiny know to trigger resize when the card size changes
     // TODO: shiny could/should do this itself (rstudio/shiny#3682)
     Card.shinyResizeObserver.observe(this.card);
+    Card.cardRemovedObserver.observe(document.body);
```

Comment:
The potential performance implications of this does not me feel it's worth rushing this in before #1009

Reply:
I agree in the sense that we should be suspicious of a MutationObserver applied to a root element like `document.body`. But I think you should look at the observer code, which does very little work, to assess the performance implications.